crates/sui-core/src/checkpoints2/mod.rs

```diff
@@ -528,6 +528,66 @@ impl CheckpointService {
         self.notify_aggregator.notify_one();
         Ok(())
     }
+
+    /// Used by internal systems that want to subscribe to checkpoints.
+    /// Returned sender will contain all checkpoints starting from(inclusive) given sequence number
+    /// CheckpointSequenceNumber::default() can be used to start from the beginning
+    pub fn subscribe_checkpoints(
+        &self,
+        from_sequence: CheckpointSequenceNumber,
+    ) -> mpsc::Receiver<(CheckpointSummary, CheckpointContents)> {
+        let (sender, receiver) = mpsc::channel(8);
+        let tailer = CheckpointTailer {
+            sender,
+            sequence: from_sequence,
+            tables: self.tables.clone(),
+            notify: self.notify_aggregator.clone(),
+        };
+        spawn_monitored_task!(tailer.run());
```

Comment:
Idea is that different components might provide different backpressure
So here I tried to focus on making it so that any tailer can tail at any rate they want, they fetch checkpoint data from disk when needed but otherwise do not provide backpressure to each other or to checkpoint process

Reply:
So this code is not intended to be used in unbounded fashion (e.g. per user request or something). Idea is that there will be few internal services that subscribe to checkpoint, and we can literally count them by number of fingers on the hand.

While in theory subscriber can notice the gap in checkpoint sequence, doing so will just take more effort on the subscriber side, not sure if we want to ask them to do so when we can just take care of it here

Reply:
Fair enough - approving